router/router.go

```
@@ -0,0 +1,340 @@
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aquasecurity/postee/data"
+	"github.com/aquasecurity/postee/dbservice"
+	"github.com/aquasecurity/postee/formatting"
+	"github.com/aquasecurity/postee/msgservice"
+	"github.com/aquasecurity/postee/outputs"
+	"github.com/aquasecurity/postee/regoservice"
+	"github.com/aquasecurity/postee/routes"
+	"github.com/aquasecurity/postee/utils"
+)
+
+const (
+	IssueTypeDefault = "Task"
+	PriorityDefault  = "High"
+
+	ServiceNowTableDefault = "incident"
+	AnonymizeReplacement   = "<hidden>"
+)
+
+type Router struct {
+	mutexScan   sync.Mutex
+	quit        chan struct{}
+	queue       chan []byte
+	ticker      *time.Ticker
+	stopTicker  chan struct{}
+	cfgfile     string
+	aquaServer  string
+	outputs     map[string]outputs.Output
+	inputRoutes map[string]*routes.InputRoute
+	templates   map[string]data.Inpteval
+}
+
+var (
+	errNoOutputs  = errors.New("there aren't started outputs")
+	initCtx       sync.Once
+	routerCtx     *Router
+	baseForTicker = time.Hour
+
+	osStat = os.Stat
+
+	requireAuthorization = map[string]bool{
+		"servicenow": true,
+		"jira":       true,
+	}
+)
+
+func Instance() *Router {
+	initCtx.Do(func() {
+		routerCtx = &Router{
+			mutexScan:   sync.Mutex{},
+			quit:        make(chan struct{}),
+			queue:       make(chan []byte, 1000),
+			outputs:     make(map[string]outputs.Output),
+			inputRoutes: make(map[string]*routes.InputRoute),
+			templates:   make(map[string]data.Inpteval),
+			stopTicker:  make(chan struct{}),
+		}
+	})
+	return routerCtx
+}
+func (ctx *Router) ReloadConfig() {
+	ctx.Terminate()
+	ctx.Start(ctx.cfgfile)
+}
+
+func (ctx *Router) Start(cfgfile string) error {
+	log.Printf("Starting Router....")
+
+	ctx.cfgfile = cfgfile
+	ctx.outputs = map[string]outputs.Output{}
+	ctx.inputRoutes = map[string]*routes.InputRoute{}
+	ctx.templates = map[string]data.Inpteval{}
+	ctx.ticker = nil
+
+	err := ctx.load()
+	if err != nil {
+		return err
```

this error will be lost.

---

fixed